presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java

```diff
@@ -67,6 +67,21 @@ public ByteCodeNode generateExpression(Signature signature, ByteCodeGeneratorCon
         ImmutableList.Builder<ByteCodeNode> defaultBucket = ImmutableList.builder();
         ImmutableSet.Builder<Object> constantValuesBuilder = ImmutableSet.builder();
 
+        boolean areSmallPrimitives = false;
+        if (values.size() < 1000 && javaType == long.class) {
+            areSmallPrimitives = true;
+            for (RowExpression testValue : values) {
+                if (testValue instanceof ConstantExpression && ((ConstantExpression) testValue).getValue() != null) {
```

Comment:
I think this logic is wrong. The loop will return true for non constants or the wrong type of constant